### PR TITLE
fix: [FFM-2184]: refactor listing pages

### PR DESCRIPTION
**Pull request by knagurski**
This PR refactors the `ListingPageTemplate` component to use more a reliable layout method, as well as implementing required `PageHeader` changes.

|Before|After|
|------|------|
|Before|After|

Refs [FFM-2184](https://harness.atlassian.net/browse/FFM-2184)

### Additions
- added `ListingPageHeading` component, styling and tests
- added `ListingPageTemplate` tests

### Changes
- refactored `ListingPageTemplate` to use a grid-based layout
- removed extraneous props from `ListingPageTemplate`
- simplified `ListingPageTemplate`
- refactored `FeatureFlagsPage` to use new template shape
- refactored `TargetsPage` to use new template shape
- refactored `SegmentsPage` to use new template shape
- refactored `EnvironmentsPage` to use new template shape
- simplified `TargetManagementHeader`

### CODE REVIEW

File: src/modules/75-cf/components/ListingPageTemplate/__tests__/ListingPageTemplate.test.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+/* eslint-disable react/display-name */
+import React, { ReactNode } from 'react'
+import { render, RenderResult, screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { TestWrapper } from '@common/utils/testUtils'
+import * as documentTitle from '@common/hooks/useDocumentTitle'
+import ListingPageTemplate, { ListingPageTemplateProps } from '../ListingPageTemplate'
+
+jest.mock('@harness/uicore', () => {
+  const fullModule = jest.requireActual('@harness/uicore')
```
Comment (chrisblakely01) [inline src/modules/75-cf/components/ListingPageTemplate/__tests__/ListingPageTemplate.test.tsx:17]: Didn't know this was a thing. Will come in handy I'm sure

**Review by chrisblakely01 — DISMISSED** [review]
Looks good! Hurray!

**Review by crystal-fong — APPROVED** [review]
looks like a good cathartic purge 👍

File: src/modules/75-cf/components/ListingPageTemplate/__tests__/ListingPageTemplate.test.tsx (hunk 1 above)
Comment (knagurski) [reply src/modules/75-cf/components/ListingPageTemplate/__tests__/ListingPageTemplate.test.tsx:17]: Yeah, I always forget it too. If you don't use it then `jest.mock` kills anything you didn't explicitly mock. I usually use `jest.spyOn` for that, but it doesn't work well will third-party stuff.

**Review by chrisblakely01 — APPROVED** [review]

**Comment by knagurski** [comment]
> looks like a good cathartic purge 👍

I flipp'n love deleting stuff :-)

**Review by crystal-fong — APPROVED** [review]

**Pull request merged**